File: lib/ts/src/internal/schema/GetPathDocumentCoordinatesPseudoJson.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { createDocumentLocatorFromYamlText } from './BuildDocumentLocatorFromYamlAst.js';
import { Coordinates, Path } from './DocumentLocators.js';

export function getPathDocumentCoordinatesPseudoJson(path: Path, document: string): Coordinates {
    try {
        return createDocumentLocatorFromYamlText(document)(path);
    } catch {
        return { row: 1, col: 1 };
    }
}

File: lib/go/internal/schema/GetPathDocumentCoordinatesPseudoJSON.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package schema

func GetPathDocumentCoordinatesPseudoJSON(path []any, document string) map[string]any {
	locator, err := createDocumentLocatorFromYAMLText(document)
	if err != nil {
		return defaultDocumentCoordinates()
	}
	return locator(append([]any{}, path...))
}
